TypeError [ERR_INVALID_URL]: Invalid URL

It looks this error returned?

`[2024-04-04 11:18:32 UTC] => [ERROR] => Newsletter: TypeError [ERR_INVALID_URL]: Invalid URL
at new NodeError (node:internal/errors:405:5)
at new URL (node:internal/url:676:13)
at Element.<anonymous> (file:///usr/src/app/utils/newsletter.js:238:29)
at LoadedCheerio.each (file:///usr/src/app/node_modules/cheerio/lib/esm/api/traversing.js:461:26)
at #injectUrlTracking (file:///usr/src/app/utils/newsletter.js:229:45)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Newsletter.renderTemplate (file:///usr/src/app/utils/newsletter.js:148:19)
at async Newsletter.send (file:///usr/src/app/utils/newsletter.js:33:60)`

_Originally posted by @jubi-git in https://github.com/ItzNotABug/ghosler/issues/36#issuecomment-2036908324_

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "ghosler",
   "description": "Send newsletter emails to your members, using your own email credentials!",
-  "version": "0.96",
+  "version": "0.97",
   "private": true,
   "main": "app.js",
   "type": "module",

--- a/utils/newsletter.js
+++ b/utils/newsletter.js
@@ -234,10 +234,18 @@ export default class Newsletter {
             let elementUrl = $(element).attr(tag);
             if (elementUrl === '#' || !elementUrl) return;
 
-            elementUrl = he.decode(elementUrl);
-            const urlHost = new URL(elementUrl).host;
+            /** @type {string | null} */
+            let urlHost = null;
 
-            if (!domainsToExclude.includes(urlHost) && !urlsToExclude.includes(elementUrl)) {
+            elementUrl = he.decode(elementUrl);
+
+            try {
+                urlHost = new URL(elementUrl).host;
+            } catch (error) {
+                logError(logTags.Newsletter, Error(`Invalid URL found: ${elementUrl}, ${error.stack}.`));
+            }
+
+            if (urlHost && !domainsToExclude.includes(urlHost) && !urlsToExclude.includes(elementUrl)) {
                 trackedLinks.add(elementUrl);
                 $(element).attr(tag, `${pingUrl}${elementUrl}`);
             }

